Core/Core/Files/View/FileDetails/FileDetailsViewController.swift

```diff
@@ -387,6 +386,27 @@ extension FileDetailsViewController: UIScrollViewDelegate {
     }
 }
 
+protocol LocalFileURLCreator {
+    func prepareLocaleURL(
+        fileName: String,
+        mimeClass: String
+    ) -> URL
+}
+
+extension LocalFileURLCreator {
+    func prepareLocaleURL(
+        fileName: String,
+        mimeClass: String
+    ) -> URL {
+        if mimeClass == "pdf" {
```

Comment:
I think the comment that was removed from here was valuable. At least for me it explained this strange logic. (The idea behind is that if the user already downloaded the file and modified if locally we don't want to download again/overwrite it, instead we give back the url pointing to the already existing file so the download will be skipped.)

Reply:
Re-added the explanation.